csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -0,0 +1,422 @@
+            }
+            else if (arrowArray is Date32Array)
+            {
+                Date32Array date32Array = (Date32Array)arrowArray;
+
+                return date32Array.GetDateTime(index);
+            }
+            else if (arrowArray is Date64Array)
+            {
+                Date64Array date64Array = (Date64Array)arrowArray;
+
+                return date64Array.GetDateTime(index);
+            }
+            else if (arrowArray is Time64Array)
+            {
+                return ((Time64Array)arrowArray).GetValue(index);
+            }
+            else if (arrowArray is TimestampArray)
+            {
+                TimestampArray timestampArray = (TimestampArray)arrowArray;
+                DateTimeOffset dateTimeOffset = timestampArray.GetTimestamp(index).Value;
+                return dateTimeOffset;
+            }
+            else if (arrowArray is StructArray)
+            {
+                StructArray structArray = (StructArray)arrowArray;
+                return SerializeToJson(structArray, index);
+            }
+            // maybe not be needed?
+            else if (arrowArray is ListArray)
+            {
+                return ((ListArray)arrowArray).GetSlicedValues(index);
+            }
+
+            return null;
+        }
+
+        static Schema TranslateSchema(TableSchema schema)
+        {
+            return new Schema(schema.Fields.Select(TranslateField), null);
+        }
+
+        static Field TranslateField(TableFieldSchema field)
+        {
+            return new Field(field.Name, TranslateType(field), field.Mode == "NULLABLE");
+        }
+
+        static IArrowType TranslateType(TableFieldSchema field)
+        {
+            // per https://developers.google.com/resources/api-libraries/documentation/bigquery/v2/java/latest/com/google/api/services/bigquery/model/TableFieldSchema.html#getType--
+
+            switch (field.Type)
+            {
+                case "INTEGER" or "INT64":
+                    return Int64Type.Default;
+                case "FLOAT" or "FLOAT64":
+                    return DoubleType.Default;
+                case "BOOL" or "BOOLEAN":
+                    return BooleanType.Default;
+                case "STRING":
+                    return StringType.Default;
+                case "BYTES":
+                    return BinaryType.Default;
+                case "DATETIME":
+                    return TimestampType.Default;
+                case "TIMESTAMP":
+                    return TimestampType.Default;
+                case "TIME":
+                    return Time64Type.Default;
+                case "DATE":
+                    return Date64Type.Default;
+                case "RECORD" or "STRUCT":
+                    // its a json string
+                    return StringType.Default;
+                // get schema cannot get precison and scale
+                case "NUMERIC" or "DECIMAL":
+                    return new Decimal128Type(38, 9);
```

Comment:
If we don't know the precision and scale of the column, the data can't get interpreted correctly. Are they always 38 and 9 or are these basically placeholders until we can produce the correct value?

Reply:
I guess the test works because 38, 9 are the precision and scale in the test data?

Reply:
Sorry I didn't see comments. The number 38 and 9 is the upper range for Decimal in bigquery. But if I remember correctly, it is more a place holder here. When read data it won't care the schema and will get the correct scale.

Reply:
I don't think this is the case, because the scale is defined by the type and not by the data in Arrow. The data just contains the digits and not the scale.

Reply:
I can try to take some tests. But I do remember here the field type only tells us that it is decimal type but with no scale and precision.

Reply:
TableFieldSchema has fields named Precision and Scale. The documentation for these says that 

Precision (maximum number of total digits in base 10) and scale (maximum number of digits in the fractional part in base 10) constraints for values of this field for NUMERIC or BIGNUMERIC. It is invalid to set precision or scale if type ≠ "NUMERIC" and ≠ "BIGNUMERIC". If precision and scale are not specified, no value range constraint is imposed on this field insofar as values are permitted by the type. Values of this NUMERIC or BIGNUMERIC field must be in this range when:

- Precision (P) and scale (S) are specified: [-10P-S + 10-S, 10P-S - 10-S]
- Precision (P) is specified but not scale (and thus scale is interpreted to be equal to zero): [-10P + 1, 10P - 1].

Acceptable values for precision and scale if both are specified:
- If type = "NUMERIC": 1 ≤ precision - scale ≤ 29 and 0 ≤ scale ≤ 9.
- If type = "BIGNUMERIC": 1 ≤ precision - scale ≤ 38 and 0 ≤ scale ≤ 38.

Acceptable values for precision if only precision is specified but not scale (and thus scale is interpreted to be equal to zero):
- If type = "NUMERIC": 1 ≤ precision ≤ 29.
- If type = "BIGNUMERIC": 1 ≤ precision ≤ 38. If scale is specified but not precision, then it is invalid.

Reply:
it has the values, but they are null when we receive them.

Reply:
I think there's some typo

- If type = "NUMERIC": 1 ≤ precision - scale ≤ 29 and 0 ≤ scale ≤ 9.
- If type = "BIGNUMERIC": 1 ≤ precision - scale ≤ 38 and 0 ≤ scale ≤ 38.

Then the precision should be

- If type = "NUMERIC": 1 ≤ precision ≤ 29 + 9(38).
- If type = "BIGNUMERIC": 1 ≤ precision - scale ≤ 38 + 38(76).

And yes the field has the properties but null when get the schema.
Actually I don't even know how to get the scale and precision for column in console. It only show "numeric" but not "numeric(p, s)". And check data_type for columns do not have those details.

Reply:
I'm thinking this issue can be traced back to the `//TODO: translate schema` comment

Reply:
Okay, so is this like the "Snowflake problem" where each chunk of Arrow data which comes back might have a precision and scale that's different than we returned as the result schema?

Reply:
Okay, I did some testing with NUMERIC (though not BIGNUMERIC) and it looks like no matter what I set as the precision and scale of the column the Arrow results are always returned as NUMERIC(38, 9). That's consistent with the bounds expressed in the documentation, and presumably the same pattern is true for BIGNUMERIC.

I think what might help things be a little less confusing would be to explicitly call this out in a comment (either directly or via a link to any documentation you might find saying so) and to have test cases for both NUMERIC(38, 9) and some other precision and scale that don't happen to exactly match the returned type of the Arrow data.